src/isomorphic/classic/class/ReactClass.js

```diff
@@ -778,8 +778,11 @@ var ReactClass = {
         // We allow auto-mocks to proceed as if they're returning null.
         if (initialState === undefined &&
             this.getInitialState._isMockFunction) {
-          // This is probably bad practice. Consider warning here and
-          // deprecating this convenience.
+          warning(
+            typeof initialState === 'object' && initialState == null && !Array.isArray(initialState),
```

Comment:
I thought the comment was meant to warn if user returns null in initialState, (also forgot about first conditional). Is the comment meant to warn AND/OR remove the mocking function? As by now, there's only two places in the repository that uses `._isMockFunction`